advanced/mathfx/src/main/java/org/arakhne/afc/math/geometry/d2/dfx/PathElement2dfx.java

```diff
@@ -249,9 +249,12 @@ public final PathElementType getType() {
 		public boolean equals(Object obj) {
 			try {
 				final PathElement2afp elt = (PathElement2afp) obj;
-				return getType() == elt.getType()
```

Comment:
Currently, the `try-catch` block is here for handling the null values for the parameter.
But in fact, Sonar rules do not recommend the use of `try-catch` in this way (that is more a programming philosophy coming for the Python language, not Java). They recommend to write equal(Object) as the following:
```java
public boolean equals(Object obj) {
    if (obj == this) {
        return true;
    }
    if (obj != null && getClass() == obj.getClass()) {
        // DO THE TEST HERE.
    }
   return false;
}
```
But I prefer:
```java
public boolean equals(Object obj) {
    if (obj == this) {
        return true;
    }
    if (getClass().isInstance(obj)) {
        // DO THE TEST HERE.
    }
   return false;
}
```
This should be applied to all equals functions.